### PR TITLE
Add pwa

### LINKED ISSUE #2033: Add PWA support
I think it would be easyd to add a Manifest file allowing to make a Progressive Web App out of FreshRSS.

For reference : https://developer.mozilla.org/en-US/docs/Web/Apps/Progressive/Installable_PWAs

On android it would be a good alternative to using EasyRSS while not being very time consuming. I think the website works well on mobile but it would be better to have it as a separated app.

### PR DESCRIPTION
**Pull request by proletarius101**
Closes https://github.com/FreshRSS/FreshRSS/issues/2033 https://github.com/FreshRSS/FreshRSS/issues/1963

Changes proposed in this pull request:

- Add a pwa manifest as defined in https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Installable_PWAs
-
-

How to test the feature manually:

1. Open FreshRSS in a mobile browser
2. Tap the `Install` button in the browser
3. An icon appears in the home screen which can open a standalone FreshRSS app

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested
- [ ] unit tests written (optional if too hard)
- [ ] documentation updated

[Additional information can be found in the documentation](https://github.com/FreshRSS/FreshRSS/tree/edge/docs/en/developers/04_Pull_requests.md).

### CODE REVIEW

**Review by Frenzie — APPROVED** [review]
lgtm

**Comment by proletarius101** [comment]
Maybe wait a sec. I'm implementing a service worker

**Comment by Frenzie** [comment]
A second? While I admit I haven't actively investigated (because we already have the likes of EasyRSS at our disposal) I always assumed that'd be a project that'd eat up at least several nights. :rofl:

**Comment by proletarius101** [comment]
Well, you may merge it now. Service worker is more complicated than I think.

File: p/themes/manifest.json [hunk 1]
```diff
@@ -2,15 +2,31 @@
...
   "icons": [
     {
       "src": "icons/favicon-256.png",
       "sizes": "256x256",
       "type": "image/png"
+    },
+    {
+      "src": "/favicon.ico",
+      "type": "image/x-icon",
+      "sizes": "16x16 64x64"
```
Comment (Frenzie) [inline p/themes/manifest.json]: It's also 32x32 and 48x48, isn't it?

**Comment by Frenzie** [comment]
We'll have to wait for @Alkarex. As per the repo settings I can only merge PRs once the CI runs successfully (which given that I or at least someone still has to port it over to GitHub Actions is currently never, hint hint if you happen to know GHA and have a half hour you'd like to spend on that :angel:).

File: p/themes/manifest.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "FreshRSS feed aggregator",
+  "short_name": "FreshRSS",
+  "start_url": "/i/",
```
Comment (Frenzie) [inline p/themes/manifest.json]:
```suggestion
  "start_url": "..",
```

**Review by Frenzie — CHANGES_REQUESTED** [review]
(withdrawing approval to prevent confusion)

File: app/layout/layout.phtml [hunk 3]
```diff
@@ -13,6 +13,7 @@ if (_t('gen.dir') === 'rtl') {
 <?php $this->renderHelper('javascript_vars'); ?>
 		</script>
 		<?= self::headScript() ?>
+		<link rel="manifest" href="<?= Minz_Url::display('/themes/manifest.json') ?>">
```
Comment (Alkarex) [inline app/layout/layout.phtml]:
```suggestion
		<link rel="manifest" href="<?= Minz_Url::display('/themes/manifest.json') ?>" />
```

File: p/themes/manifest.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    }
+  ],
+  "categories": [
+    "news"
+  ],
+  "display": "standalone",
+  "background_color": "#fff",
+  "theme_color": "#0062be",
+  "description": "FreshRSS is a self-hosted RSS feed aggregator."
+}
```
Comment (Alkarex) [inline p/themes/manifest.json]:
```suggestion
}

```

File: p/themes/manifest.json (hunk 1 above)
Comment (proletarius101) [reply p/themes/manifest.json]: fixed

File: p/themes/manifest.json (hunk 2 above)
Comment (proletarius101) [reply p/themes/manifest.json]: fixed

File: p/themes/manifest.json [hunk 5]
```diff
@@ -3,7 +3,6 @@
   "short_name": "FreshRSS",
   "start_url": "../i/",
   "scope": "../",
-  "orientation": "any",
```
Comment (Frenzie) [inline p/themes/manifest.json]: Is leaving this out equivalent to any? Enforcing portrait would strike me as undesirable.

**Review by Frenzie — APPROVED** [review]
lgtm

File: p/themes/manifest.json (hunk 5 above)
Comment (proletarius101) [reply p/themes/manifest.json]: As most apps, the orientation is controlled by the system rather than the app

Comment (Frenzie) [reply p/themes/manifest.json]: It's mainly that I don't know what prompted you to include it in the first place. I assumed there was a reason for it. ;-)

**Comment by Alkarex** [comment]
> We'll have to wait for @Alkarex. As per the repo settings I can only merge PRs once the CI runs successfully (which given that I or at least someone still has to port it over to GitHub Actions is currently never, hint hint if you happen to know GHA and have a half hour you'd like to spend on that 👼).

CI fixed https://github.com/FreshRSS/FreshRSS/pull/3920

**Comment by Alkarex** [comment]
I will have to look at bit more about it. I am not (yet) convinced of the advantages. This PR is probably fine as it does not "cost" much, but the related https://github.com/FreshRSS/FreshRSS/pull/3896 is much larger, with dependencies, and for an unclear advantage. Additional input welcome.

**Comment by proletarius101** [comment]
> I will have to look at bit more about it. I am not (yet) convinced of the advantages. This PR is probably fine as it does not "cost" much, but the related #3896 is much larger, with dependencies, and for an unclear advantage. Additional input welcome.

This PR doesn't rely on https://github.com/FreshRSS/FreshRSS/pull/3896. Vise versa. So they can be merged independently.

- This PR is to enable [Installability](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Introduction#installability), making it native to the system
- The service worker is for [Network independence](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Introduction#network_independence), which makes reading pre-cached articles offline possible.
  - It also provides transition between poor connectivity and good connectivity

**Comment by Frenzie** [comment]
A potential advantage of a PWA (not in the current poc unless I missed some more recent changes) would be that syncing read/unread status and the like could be handled by the service worker, without having interrupted or absent connections throwing spanners in the works.

**Comment by Alkarex** [comment]
Ok, let's give this one a try. Yes @Frenzie , regarding the second PR, having a way to work partially offline would be interesting to look into, but that should probably require reworking the UI rendering to be more client side, potentially taking advantage of our API like an app. But the question is then whether a full app is not better for this role and there are a few good clients already.

**Comment by Frenzie** [comment]
Right, I never looked into it myself because we already have EasyRSS and many others. But you can also build up to something progressively. I'm ambivalent towards it. :-) https://gomakethings.com/service-worker-strategies/

**Comment by math-GH** [comment]
I love this PWA feature. Tested it with Mozilla Firefox on Android.

**Comment by dminca** [comment]
tested 1.29.1 on Vanadium (Android) and it's not working as expected, it's just saved as a bookmark on the home.

**Comment by Frenzie** [comment]
It's not that very different from a bookmark anyway, but I see the two separate options.
IMG_20260903_211558

**Comment by dminca** [comment]
This is the difference

IMG_20260903_214258

**Fig. 1** - example with working PWA (in this case a different app called Memos)

----

IMG_20260903_214102

**Fig. 2** - FreshRSS not working PWA

It seems FreshRSS is not available as a PWA

**Comment by Frenzie** [comment]
It works in Chrome and Vivaldi. You'd have to try a few other browsers to see if it's something about that specific installation or something in combination with Vanadium. I assume you mean you have seen the option to "install" in Vanadium before?

**Comment by dminca** [comment]
I just downloaded Vivaldi on Android from the Play Store.

Only have the option to _add page to > Home screen_ and this mimics the same behaviour, still no PWA 🤔 (as seen in Fig. 2 I shared above)

> I assume you mean you have seen the option to "install" in Vanadium before?

yes, it's Fig. 1 from the screenshots I shared above.

**Comment by Frenzie** [comment]
It's probably best to open an issue. I'm out of ideas for the moment. Maybe it's some kind of reverse proxy issue?

**Comment by dminca** [comment]
No worries, will do that. Thanks a lot so far :)

LE: done, carded #9258

**Pull request merged**